neo4j/src/main/java/org/apache/metamodel/neo4j/Neo4jDataContext.java

```diff
@@ -0,0 +1,336 @@
+            return allNodesPerLabel;
+        } catch (JSONException e) {
+            logger.error("Error occured in parsing JSON while detecting the nodes for a label: " + label, e);
+            throw new IllegalStateException(e);
+        }
+    }
+
+    private List<String> getAllPropertiesPerNode(JSONObject node) {
+        List<String> properties = new ArrayList<String>();
+        properties.add("_id");
+
+        String propertiesEndpoint;
+        try {
+            propertiesEndpoint = node.getString("properties");
+
+            String allPropertiesPerNodeJsonString = _requestWrapper.executeRestRequest(new HttpGet(propertiesEndpoint));
+
+            JSONObject allPropertiesPerNodeJsonObject = new JSONObject(allPropertiesPerNodeJsonString);
+            for (int j = 0; j < allPropertiesPerNodeJsonObject.length(); j++) {
+                JSONArray propertiesJsonArray = allPropertiesPerNodeJsonObject.names();
+                for (int k = 0; k < propertiesJsonArray.length(); k++) {
+                    String property = propertiesJsonArray.getString(k);
+                    properties.add(property);
+                }
+            }
+            return properties;
+        } catch (JSONException e) {
+            logger.error("Error occured in parsing JSON while detecting the properties of a node: " + node, e);
+            throw new IllegalStateException(e);
+        }
+    }
+
+    @Override
+    protected DataSet materializeMainSchemaTable(Table table, Column[] columns, int firstRow, int maxRows) {
+        if ((columns != null) && (columns.length > 0)) {
+            Neo4jDataSet dataSet = null;
+            try {
+                String selectQuery = Neo4jCypherQueryBuilder.buildSelectQuery(table, columns, firstRow, maxRows);
+                String responseJSONString = _requestWrapper.executeCypherQuery(selectQuery);
+                JSONObject resultJSONObject = new JSONObject(responseJSONString);
+                final SelectItem[] selectItems = MetaModelHelper.createSelectItems(columns);
+                dataSet = new Neo4jDataSet(selectItems, resultJSONObject);
+            } catch (JSONException e) {
+                logger.error("Error occured in parsing JSON while materializing the schema: ", e);
+                throw new IllegalStateException(e);
+            }
+
+            return dataSet;
+        } else {
+            logger.error("Encountered null or empty columns array for materializing main schema table.");
+            throw new IllegalArgumentException("Columns cannot be null or empty array");
+        }
+    }
+
+    @Override
+    protected DataSet materializeMainSchemaTable(Table table, Column[] columns, int maxRows) {
+        return materializeMainSchemaTable(table, columns, 1, maxRows);
+    }
+
+    @Override
+    protected org.apache.metamodel.data.Row executePrimaryKeyLookupQuery(Table table, List<SelectItem> selectItems,
+            Column primaryKeyColumn, Object keyValue) {
+        return null;
```

Comment:
If we don't have an implementation here then I suggest not overriding it since it does nothing.